### ISSUE TITLE
💅 False positive lint/nursery/useExplicitType on module-level assignment when RHS is not a constant

### ISSUE THREAD

**Issue by sterliakov** [biomejs/biome#5932]
### Environment information

```bash
CLI:
  Version:                      2.0.0-beta.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.18.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/10.8.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         /home/stas/Documents/Work/simpleem-fe/biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  true
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noBannedTypes
    complexity/noEmptyTypeParameters
    complexity/noExtraBooleanCast
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessEmptyExport
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noWith
    complexity/useArrowFunction
    complexity/useFlatMap
    complexity/useLiteralKeys
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noFlatMapIdentity
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noNodejsModules
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUndeclaredVariables
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noUselessContinue
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useHookAtTopLevel
    correctness/useIsNan
    correctness/useValidForDirection
    correctness/useYield
    nursery/noDuplicateElseIf
    nursery/noDuplicateProperties
    nursery/noTemplateCurlyInString
    nursery/noUnknownPseudoClass
    nursery/noUnknownPseudoElement
    nursery/noUnknownTypeSelector
    nursery/useAdjacentOverloadSignatures
    nursery/useCollapsedIf
    nursery/useComponentExportOnlyModules
    nursery/useExplicitType
    performance/noAccumulatingSpread
    security/noBlankTarget
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    style/noUselessElse
    style/useCollapsedElseIf
    style/useForOf
    style/useShorthandAssign
    style/useSingleCaseStatement
    suspicious/noApproximativeNumericConstant
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConstEnum
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noEmptyBlock
    suspicious/noEmptyInterface
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noPrototypeBuiltins
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noThenProperty
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/useAwait
    suspicious/useDefaultSwitchClauseLast
    suspicious/useGetterReturn
    suspicious/useIsArray
    suspicious/useNamespaceKeyword
    suspicious/useValidTypeof

Workspace:
  Open Documents:               0
```

### Rule name

lint/nursery/useExplicitType

### Playground link

https://next.biomejs.dev/playground/?indentStyle=space&indentWidth=4&lintRules=all&enabledAssist=false&code=YwBvAG4AcwB0ACAAZgBvAG8AOgAgAHMAdAByAGkAbgBnACAAfAAgAG4AdQBsAGwAIAA9ACAAbABvAGMAYQBsAFMAdABvAHIAYQBnAGUALgBnAGUAdABJAHQAZQBtACgAIgBmAG8AbwAiACkAOwAKAA%3D%3D&language=ts

### Expected result

No error, the variable is explicitly annotated with a type.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by sterliakov** [comment]
Since this is now confirmed, I'll try to submit a patch today.
